ansible_collections/arista/avd/roles/eos_designs/python_modules/inband_management/avdstructuredconfig.py

```diff
@@ -135,38 +169,73 @@ def route_maps(self) -> list | None:
         if not self.shared_utils.underlay_filter_redistribute_connected:
             return None
 
+        if not self._inband_mgmt_ipv6_parent:
+            return None
+
+        sequence_numbers = [
+            {
+                "sequence": (index + 1) * 10,
+                "action": f"permit {subnet['ipv6']}",
+            }
+            for index, subnet in enumerate(self._inband_management_parent_vlans.values())
+        ]
         return [
             {
-                "name": "RM-CONN-2-BGP",
-                "sequence_numbers": [
-                    {
-                        # sequence 10 is set in underlay so avoid setting it here
-                        "sequence": 20,
-                        "type": "permit",
-                        "match": ["ip address prefix-list PL-L2LEAF-INBAND-MGMT"],
-                    }
-                ],
+                "name": "IPv6-PL-L2LEAF-INBAND-MGMT",
+                "sequence_numbers": sequence_numbers,
             }
         ]
 
+    @cached_property
+    def route_maps(self) -> list | None:
+        if not self._inband_management_parent_vlans:
```

Comment:
@ClausHolbechArista would not we want this in a separate file / Mixin so it is easier to pick up for our impending refactoring?